Entity generation fails creating relationship to nonexistent entity

##### **Overview of the issue**

New code in JHipster 4.0.5 fails when creating an entity that has a relationship to another entity that does not exist yet.

Stack trace:
events.js:161
      throw er; // Unhandled 'error' event
      ^

TypeError: Cannot read property 'angularJSSuffix' of undefined
    at constructor.<anonymous> (/Applications/BugTracker2/node_modules/generator-jhipster/generators/entity/index.js:585:71)
    at Array.forEach (native)
    at constructor.loadInMemoryData (/Applications/BugTracker2/node_modules/generator-jhipster/generators/entity/index.js:523:54)
    at Object.<anonymous> (/Applications/BugTracker2/node_modules/yeoman-generator/lib/base.js:439:23)
    at /Applications/BugTracker2/node_modules/yeoman-generator/node_modules/run-async/index.js:25:25
    at /Applications/BugTracker2/node_modules/yeoman-generator/node_modules/run-async/index.js:24:19
    at /Applications/BugTracker2/node_modules/yeoman-generator/lib/base.js:440:9
    at runCallback (timers.js:651:20)
    at tryOnImmediate (timers.js:624:5)
    at processImmediate [as _immediateCallback] (timers.js:596:5)

##### **Reproduce the error**

- Create a new JHipster project using MySQL for prod/dev
- Create a new entity with a relationship to a nonexistent entity
- Stack trace above shows in terminal

##### **Suggest a Fix**

This line seems to be the problem:
https://github.com/jhipster/generator-jhipster/blob/master/generators/entity/index.js#L585

Perhaps a null check on the 'otherEntityData' variable?

##### **JHipster Version(s)**

4.0.5

##### **JHipster configuration**

Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
bug-tracker-2@0.0.0 /Applications/BugTracker2
├── UNMET PEER DEPENDENCY @angular/compiler@2.4.7
├── UNMET PEER DEPENDENCY @angular/core@2.4.7
└── generator-jhipster@4.0.5 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

```yaml
{
  "generator-jhipster": {
    "jhipsterVersion": "4.0.5",
    "baseName": "BugTracker2",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "session",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "rememberMeKey": "dbfae88f50b8a3deab7cf0e5e9e7fa7b42f507aa",
    "clientFramework": "angular2",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en"
    ]
  }
}
```


##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

Label.json

```yaml
{
    "fluentMethods": true,
    "relationships": [],
    "fields": [
        {
            "fieldName": "label",
            "fieldType": "String",
            "fieldValidateRules": [
                "required",
                "minlength"
            ],
            "fieldValidateRulesMinlength": "3"
        }
    ],
    "changelogDate": "20170220192625",
    "dto": "no",
    "service": "no",
    "entityTableName": "label",
    "pagination": "no"
}

```

Mama.json

```yaml
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipName": "ticket",
            "otherEntityName": "ticket",
            "relationshipType": "many-to-many",
            "ownerSide": false,
            "otherEntityRelationshipName": "mama"
        }
    ],
    "fields": [
        {
            "fieldName": "name",
            "fieldType": "String"
        }
    ],
    "changelogDate": "20170220200427",
    "dto": "no",
    "service": "no",
    "entityTableName": "mama",
    "pagination": "no"
}

```

Project.json

```yaml
{
    "fluentMethods": true,
    "relationships": [],
    "fields": [
        {
            "fieldName": "name",
            "fieldType": "String"
        }
    ],
    "changelogDate": "20170220191158",
    "dto": "no",
    "service": "no",
    "entityTableName": "project",
    "pagination": "no"
}

```

Ticket.json

```yaml
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipName": "project",
            "otherEntityName": "project",
            "relationshipType": "many-to-one",
            "otherEntityField": "name"
        },
        {
            "relationshipName": "assignedTo",
            "otherEntityName": "user",
            "relationshipType": "many-to-one",
            "otherEntityField": "login",
            "ownerSide": true,
            "otherEntityRelationshipName": "ticket"
        },
        {
            "relationshipName": "label",
            "otherEntityName": "label",
            "relationshipType": "many-to-many",
            "otherEntityField": "label",
            "ownerSide": true,
            "otherEntityRelationshipName": "ticket"
        }
    ],
    "fields": [
        {
            "fieldName": "title",
            "fieldType": "String",
            "fieldValidateRules": [
                "required"
            ]
        },
        {
            "fieldName": "description",
            "fieldType": "String"
        },
        {
            "fieldName": "dueDate",
            "fieldType": "LocalDate"
        },
        {
            "fieldName": "done",
            "fieldType": "Boolean"
        }
    ],
    "changelogDate": "20170220193219",
    "dto": "no",
    "service": "no",
    "entityTableName": "ticket",
    "pagination": "no"
}

```


##### **Browsers and Operating System**

java version "1.8.0_25"
Java(TM) SE Runtime Environment (build 1.8.0_25-b17)
Java HotSpot(TM) 64-Bit Server VM (build 25.25-b02, mixed mode)

git version 2.9.3 (Apple Git-75)

node: v7.5.0

npm: 4.1.2

bower: 1.8.0

gulp:
[15:44:32] CLI version 1.2.2

yeoman: 1.8.5

yarn: 0.20.0

Docker version 1.13.1, build 092cba3

docker-compose version 1.11.1, build 7c5d5e4


##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

Included in print out above

##### **Browsers and Operating System**

MacOS Sierra 10.12.1

- [x] Checking this box is mandatory (this is just to show you read everything)

I just tried making the null check locally and then I get a new error:

ERROR in ./src/main/webapp/app/entities/label/label-dialog.component.ts
Module not found: Error: Can't resolve '../ticket' in '/Applications/BugTracker4/src/main/webapp/app/entities/label'
 @ ./src/main/webapp/app/entities/label/label-dialog.component.ts 17:15-35
 @ ./src/main/webapp/app/entities/label/index.ts
 @ ./src/main/webapp/app/entities/label/label.module.ts
 @ ./src/main/webapp/app/entities/entity.module.ts
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts

ERROR in [at-loader] src/main/webapp/app/entities/label/label-dialog.component.ts:11:39 
    TS2307: Cannot find module '../ticket'.

ERROR in [at-loader] src/main/webapp/app/entities/label/label.model.ts:1:24 
    TS2307: Cannot find module '../ticket'.
Child html-webpack-plugin for "index.html":
    [./node_modules/html-webpack-plugin/lib/loader.js!./src/main/webapp/index.html] ./~/html-webpack-plugin/lib/loader.js!./src/main/webapp/index.html 1.22 kB {0} [built]
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

--- a/generators/entity/index.js
+++ b/generators/entity/index.js
@@ -582,7 +582,7 @@ module.exports = EntityGenerator.extend({
 
                 if (_.isUndefined(relationship.otherEntityAngularName)) {
                     if (relationship.otherEntityNameCapitalized !== 'User') {
-                        var otherEntityAngularSuffix = otherEntityData.angularJSSuffix || '';
+                        var otherEntityAngularSuffix = otherEntityData ? otherEntityData.angularJSSuffix || '' : '';
                         relationship.otherEntityAngularName = _.upperFirst(relationship.otherEntityName) + _.upperFirst(_.camelCase(otherEntityAngularSuffix));
                     } else {
                         relationship.otherEntityAngularName = 'User';

